support/routes/v2/isobot/ops/botService.ts

```diff
@@ -91,6 +170,276 @@ class BotService {
+      )
+      response.push("This is *AGENCY fault*!")
+    } else if (!cnameRecord && !intermediateRecords && !redirectionRecords) {
+      // Everything is gone
+      response.push(
+        `Jialat, the DNS for \`${domain}\` seems to be gone, and our indirection layer does not seem to be configured correctly.`
+      )
+      response.push(
+        "If this site is supposed to be live, then this is *both OUR and AGENCY fault*!"
+      )
+    } else {
+      // Some weird configuration
+      response.push(
+        `Wah, I don't know how to handle \`${domain}\` sia, might need to manually check.`
+      )
+      response.push(
+        `-   A records checker: https://dnschecker.org/#A/${domain}`
+      )
+      response.push(
+        `-   CNAME records checker: https://dnschecker.org/#CNAME/${domain}`
+      )
+    }
+
+    response.push(
+      `-   \`${domain}\` points to ${
+        cnameRecord ? `\`${cnameRecord}\`` : "no CNAME records"
+      }`
+    )
+
+    if (cnameRecord || indirectionDomain) {
+      response.push(
+        `-   \`${cnameRecord || indirectionDomain}\` points to ${
+          intermediateRecords
+            ? `\`${intermediateRecords.join(", ")}\``
+            : "nothing"
+        }`
+      )
+    }
+
+    if (redirectionDomain !== cnameDomain) {
+      response.push(
+        `-   \`${redirectionDomain}\` points to ${
+          redirectionRecords
+            ? `\`${redirectionRecords.join(", ")}\``
+            : "no A records"
+        }`
+      )
+    }
+
+    return this.getSlackMessage(response)
+  }
+
+  dnsChecker(payload: SlackPayload) {
+    // Step 1: Get the domain name provided by the user
+    const { user_name: user, channel_name: channel, text: domain } = payload
+    logger.info({
+      message: "DNS check requested",
+      meta: {
+        method: "dnsChecker",
+        user,
+        channel,
+        domain,
+      },
+    })
+
+    return this.checkCname(domain)
+      .andThen((cname) => {
+        // Original domain does not have a CNAME record, check if the www
+        // version has a valid CNAME record
+        if (!cname && !domain.startsWith("www.")) {
+          const cnameDomain = `www.${domain}`
+          return ResultAsync.combine([
+            okAsync(cnameDomain),
+            this.checkCname(cnameDomain),
+          ])
+        }
+
+        return ResultAsync.combine([okAsync(domain), okAsync(cname)])
+      })
+      .andThen(([cnameDomain, cnameRecord]) => {
+        // Original and www version of the domain do not have a CNAME record,
+        // check if our indirection domain is still correct
+        if (!cnameRecord) {
+          const indirectionDomain = domain.startsWith("www.")
+            ? `${domain
+                .slice(4)
+                .replaceAll(".", "-")}.${DNS_INDIRECTION_DOMAIN}`
+            : `${domain.replaceAll(".", "-")}.${DNS_INDIRECTION_DOMAIN}`
+
+          return ResultAsync.combine([
+            okAsync(null),
```

Comment:
i believe this is misplaced

Reply:
Hmm no, this one is explicit to be null because the cnameDomain does not actually exist, so we know that it is not valid.